File: admin-web/src/stores/report.ts
import { defineStore } from 'pinia'
import { computed, ref } from 'vue'

import { fetchBoardTasks, fetchDashboardSummary } from '@/api/reports'

export interface DashboardSummaryResponse {
  metric_version: string
  snapshot_time: string
  sla_status: string
  contract_execution_rate: string
  actual_receipt_today: string
  actual_payment_today: string
  inventory_turnover_30d: string
  threshold_alert_count: number
  message: string
}

export interface BoardTaskItem {
  biz_type: string
  biz_id: number
  title: string
  status: string
  contract_id: number | null
  contract_no: string | null
  related_order_id: number | null
  created_at: string | null
}

export interface BoardTasksResponse {
  metric_version: string
  snapshot_time: string
  sla_status: string
  pending_supplement_count: number
  validation_failed_count: number
  qty_done_not_closed_count: number
  pending_supplement_items: BoardTaskItem[]
  validation_failed_items: BoardTaskItem[]
  qty_done_not_closed_items: BoardTaskItem[]
  message: string
}

export const useReportStore = defineStore('report', () => {
  const dashboard = ref<DashboardSummaryResponse | null>(null)
  const board = ref<BoardTasksResponse | null>(null)
  const dashboardLoading = ref(false)
  const boardLoading = ref(false)
  const dashboardError = ref('')
  const boardError = ref('')

  async function loadDashboard() {
    dashboardLoading.value = true
    dashboardError.value = ''
    try {
      dashboard.value = await fetchDashboardSummary()
    } catch (error) {
      dashboardError.value = error instanceof Error ? error.message : '仪表盘加载失败'
    } finally {
      dashboardLoading.value = false
    }
  }

  async function loadBoard() {
    boardLoading.value = true
    boardError.value = ''
    try {
      board.value = await fetchBoardTasks()
    } catch (error) {
      boardError.value = error instanceof Error ? error.message : '业务看板加载失败'
    } finally {
      boardLoading.value = false
    }
  }

  const anomalyDistribution = computed(() => {
    if (!board.value) {
      return []
    }
    const total =
      board.value.pending_supplement_count +
      board.value.validation_failed_count +
      board.value.qty_done_not_closed_count
    return [
      {
        key: 'pending',
        label: '待补录金额',
        value: board.value.pending_supplement_count,
        ratio: total > 0 ? board.value.pending_supplement_count / total : 0,
      },
      {
        key: 'failed',
        label: '校验失败',
        value: board.value.validation_failed_count,
        ratio: total > 0 ? board.value.validation_failed_count / total : 0,
      },
      {
        key: 'qtydone',
        label: '数量履约完成未关闭',
        value: board.value.qty_done_not_closed_count,
        ratio: total > 0 ? board.value.qty_done_not_closed_count / total : 0,
      },
    ]
  })

  return {
    anomalyDistribution,
    board,
    boardError,
    boardLoading,
    dashboard,
    dashboardError,
    dashboardLoading,
    loadBoard,
    loadDashboard,
  }
})

File: backend/app/schemas/report.py
from __future__ import annotations

from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, Field


class DashboardSummaryResponse(BaseModel):
    metric_version: str
    snapshot_time: datetime
    sla_status: str
    contract_execution_rate: Decimal
    actual_receipt_today: Decimal
    actual_payment_today: Decimal
    inventory_turnover_30d: Decimal
    threshold_alert_count: int
    message: str


class BoardTaskItem(BaseModel):
    biz_type: str
    biz_id: int
    title: str
    status: str
    contract_id: int | None = None
    contract_no: str | None = None
    related_order_id: int | None = None
    created_at: datetime | None = None


class BoardTasksResponse(BaseModel):
    metric_version: str
    snapshot_time: datetime
    sla_status: str
    pending_supplement_count: int
    validation_failed_count: int
    qty_done_not_closed_count: int
    pending_supplement_items: list[BoardTaskItem] = Field(default_factory=list)
    validation_failed_items: list[BoardTaskItem] = Field(default_factory=list)
    qty_done_not_closed_items: list[BoardTaskItem] = Field(default_factory=list)
    message: str


class LightReportOverviewResponse(BaseModel):
    metric_version: str
    snapshot_time: datetime
    sla_status: str
    actual_receipt_today: Decimal
    actual_payment_today: Decimal
    inbound_qty_today: Decimal
    outbound_qty_today: Decimal
    abnormal_count: int
    pending_supplement_count: int
    validation_failed_count: int
    qty_done_not_closed_count: int
    message: str


class AdminMultiDimReportRow(BaseModel):
    dimension: str
    dimension_value: str
    receipt_net_amount: Decimal
    payment_net_amount: Decimal
    net_cashflow: Decimal
    receipt_doc_count: int
    payment_doc_count: int
    pending_supplement_count: int
    refund_pending_review_count: int


class AdminMultiDimReportResponse(BaseModel):
    metric_version: str
    snapshot_time: datetime
    sla_status: str
    group_by: str
    filters: dict[str, str | None]
    total_receipt_net_amount: Decimal
    total_payment_net_amount: Decimal
    total_net_cashflow: Decimal
    rows: list[AdminMultiDimReportRow] = Field(default_factory=list)
    message: str


class AdminMultiDimExportTaskCreateRequest(BaseModel):
    metric_version: str | None = Field(default=None, max_length=16)
    group_by: str = Field(default="contract_direction")
    contract_direction: str | None = Field(default=None)
    doc_status: str | None = Field(default=None, max_length=16)
    refund_status: str | None = Field(default=None, max_length=16)
    date_from: date | None = None
    date_to: date | None = None


class AdminMultiDimExportTaskItem(BaseModel):
    id: int
    report_code: str
    report_name: str
    status: str
    export_format: str
    metric_version: str
    filters: dict[str, str | None]
    file_name: str | None = None
    requested_by: str
    requested_role_code: str
    requested_company_id: str | None = None
    retry_count: int
    download_count: int
    error_message: str | None = None
    finished_at: datetime | None = None
    created_at: datetime
    updated_at: datetime


class AdminMultiDimExportTaskCreateResponse(BaseModel):
    task: AdminMultiDimExportTaskItem
    message: str


class AdminMultiDimExportTaskListResponse(BaseModel):
    items: list[AdminMultiDimExportTaskItem] = Field(default_factory=list)
    message: str
